plugins/filter/to_prettytable.py

```diff
@@ -0,0 +1,382 @@
+    """
+    key_map = {}
+    reverse_key_map = {}
+
+    first_dict = data[0]
+    for orig_key in first_dict.keys():
+        # Store string version of the key
+        str_key = to_text(orig_key)
+        key_map[str_key] = orig_key
+        # Also store lowercase version for case-insensitive lookups
+        reverse_key_map[str_key.lower()] = orig_key
+
+    return key_map, reverse_key_map
+
+
+def _configure_alignments(table, field_names, column_alignments):
+    """Configure column alignments for the table.
+
+    Args:
+        table: The PrettyTable instance to configure
+        field_names: List of field names to align
+        column_alignments: Dict of column alignments
+    """
+    valid_alignments = {"left", "center", "right", "l", "c", "r"}
+
+    if not isinstance(column_alignments, dict):
+        return
+
+    for col_name, alignment in column_alignments.items():
+        if col_name in field_names:
+            # We already validated alignment is a string and a valid value in the main function
+            # Just apply it here
+            alignment = alignment.lower()
+            table.align[col_name] = alignment[0]
+
+
+def to_prettytable(data, *args, **kwargs):
+    """Convert a list of dictionaries to an ASCII table.
+
+    Args:
+        data: List of dictionaries to format
+        *args: Optional list of column names to specify column order
+        **kwargs: Optional keyword arguments:
+            - column_order: List of column names to specify the order
+            - header_names: List of custom header names
+            - column_alignments: Dict of column alignments (left, center, right)
+
+    Returns:
+        String containing the ASCII table
+    """
+    if not HAS_PRETTYTABLE:
+        raise AnsibleFilterError(
+            'You need to install "prettytable" Python module to use this filter'
+        )
+
+    # Handle empty data
+    if not data:
+        return "++\n++"
+
+    # === Input validation ===
+    # Validate list type
+    if not isinstance(data, list):
+        raise AnsibleFilterError(_type_error(data, "a list of dictionaries"))
+
+    # Validate dictionary items
+    if data and not all(isinstance(item, dict) for item in data):
+        invalid_item = next((item for item in data if not isinstance(item, dict)), None)
+        raise AnsibleFilterError(_type_error(invalid_item, "all items in the list to be dictionaries"))
+
+    # Get the maximum number of fields in the first dictionary
+    max_fields = len(data[0].keys())
+
+    # === Process column order ===
+    # Handle both positional and keyword column_order
+    column_order = kwargs.pop('column_order', None)
+
+    # Check for conflict between args and column_order
+    if args and column_order is not None:
+        raise AnsibleFilterError("Cannot use both positional arguments and the 'column_order' keyword argument")
+
+    # Use positional args if provided
+    if args:
+        column_order = list(args)
```

Comment:
This could be simpler as:
```suggestion
    if args and column_order is not None:
        raise AnsibleFilterError("Cannot use both positional arguments and the 'column_order' keyword argument")

    column_order = args or kwargs.pop('column_order', None)
```
(btw, `args` is already a list, no need to put it through `list()`)

Reply:
Sorry, it looks like the suggested approach does not work as expected. First, you cannot check `column_order` value in line 258 before you really set this variable in line 261. It returns an error of undefined variable this way. Second, removing explicit list in line 266 (from initial code) breaks integration tests and makes it impossible to check the args correctly. If I remove explicit list from line 266, I get the following error in tests:
```
TASK [filter_to_prettytable : set_fact] ****************************************
fatal: [testhost]: FAILED! => {"msg": "Expected a list of column names, got a tuple"}
...ignoring

TASK [filter_to_prettytable : Verify error message for non-string values in positional args] ***
fatal: [testhost]: FAILED! => {
    "assertion": "'Expected a string for column name, got a int' in failure_result.msg",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```